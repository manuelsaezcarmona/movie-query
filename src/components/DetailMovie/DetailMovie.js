/* eslint-disable max-len */

import styles from './styles.module.css';

export default function DetailMovie() {
  return (
    <main className="main-container">
      <div className={styles['film-detail']}>
        <header className={styles['film-header']}>
          <h2 className={styles['film-header__title']}>Jack Reacher: Nunca vuelvas atrás</h2>
          <p className={styles['film-header__tagline']}>Nunca cedas. Nunca te des por vencido. Nunca vuelvas.</p>
          <p className={styles['film-header__valuation']}>
            Valoracion: <span className={styles['film-header__rate']}>5.9</span>
          </p>
        </header>
        <section className={styles['film-body']}>
          <img
            className={styles['film-body__image']}
            src="https://image.tmdb.org/t/p/w500//gsIaANx3PVGMkzHfwdCOMUP0o7U.jpg"
            alt="Poster de la pelicula"
          />

          <div className={styles['film-body__data']}>
            <h3 className={styles['film-body__title']}>Jack Reacher: Never Go Back</h3>
            <p className={styles['film-body__release']}>
              Fecha de Estreno: <span className={styles['film-body__date']}>12-09-2009</span>
            </p>
            <p className={styles['film-body__sipnosis']}>
              La Mayor Susan Turner, líder de la antigua unidad militar de Reacher, es falsamente acusada de traición.
              Jack Reacher tendrá que sacarla de prisión y descubrir la verdad detrás de una conspiración gubernamental
              para limpiar sus nombres y salvar sus vidas. Durante el escape, Reacher descubrirá un secreto de su pasado
              que podría cambiar su vida para siempre. Secuela de Jack Reacher (2012).
            </p>
          </div>
        </section>
      </div>
    </main>
  );
}
