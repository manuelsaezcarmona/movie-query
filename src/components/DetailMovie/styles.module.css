.film-detail {
  display: flex;
  flex-direction: column;
}

.film-header {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 35vh;
  margin: 0;
  position: relative;
  overflow: hidden;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  padding: 0 1em;
}
.film-header::before {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  overflow: hidden;
  top: 0;
  left: 0;
  background-image: linear-gradient(rgba(0, 0, 0, 0.4), rgba(0, 0, 0, 0.4)),
    url('https://image.tmdb.org/t/p/w500/ww1eIoywghjoMzRLRIcbJLuKnJH.jpg');
  z-index: -1;
  background-size: cover;
  background-position: left center;
  background-repeat: no-repeat;
  transform: skewY(-2.2deg);
  transform-origin: 0 0;
  /*chrome antialias fix*/
  -webkit-backface-visibility: hidden;
}

.film-header__title,
.film-header__tagline,
.film-header__valuation {
  color: var(--navbar-background-color);
}
.film-header__title {
  font-size: 1.2em;
  margin-bottom: 0.5vh;
}

.film-header__tagline {
  font-size: small;
  margin-bottom: 0.5vh;
}

.film-body {
  display: flex;
  flex-direction: column;
}

.film-body__image {
  margin-top: -9vh;

  max-width: 50%;
  max-height: auto;
  object-fit: contain;
  border: 2px white solid;
  align-self: center;
  box-shadow: var(--shadow);
}
.film-body__title,
.film-body__release {
  margin: 1vh 0;
  text-align: center;
}

.film-body__sipnosis {
  line-height: 1.3;
}

@media (min-width: 769px) {
  .film-header {
    height: 40vh;
    align-items: center;
  }
  .film-body__image {
    margin-left: 8%;
    max-width: 30%;
  }
  .film-header__title {
    font-size: 2em;
  }
  .film-header__tagline,
  .film-header__valuation {
    font-size: 1.1em;
  }
  .film-body {
    flex-direction: row;
    justify-content: center;
  }
  .film-body__data {
    padding-left: 2em;
  }
}
