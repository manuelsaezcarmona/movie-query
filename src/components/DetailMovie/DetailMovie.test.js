import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import DetailMovie from './DetailMovie';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  user: {
    username: 'test',
    favourites: []
  },
  movie: {
    page: 1,
    movies: [],
    activemovie: {},
    totalpages: 3,
    detailedmovie: {}
  },
  settings: {
    currentpage: 1,
    searchfilm: 'Avengers'
  }
};

const store = mockStore(initialState);
store.dispatch = jest.fn();

jest.mock('../../redux/actions/movie', () => ({
  startGetMovieById: jest.fn()
}));

describe('Given the DetailMovie Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <DetailMovie />
      </Provider>
    );
  });
  test('The component is rendered (take a snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <DetailMovie />
      </Provider>
    );
    expect(asFragment(<DetailMovie />)).toMatchSnapshot();
  });
});
