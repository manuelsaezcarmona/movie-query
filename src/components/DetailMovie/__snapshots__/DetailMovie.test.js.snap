// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the DetailMovie Component The component is rendered (take a snapshot) 1`] = `
<DocumentFragment>
  <main
    class="main-container"
  >
    <div
      class="film-detail"
    >
      <header
        class="film-header"
      >
        <img
          alt="background undefined"
          class="film-header__background"
          src="https://image.tmdb.org/t/p/w780undefined"
        />
        <h2
          class="film-header__title"
        />
        <p
          class="film-header__tagline"
        >
          .
        </p>
        <p
          class="film-header__valuation"
        >
          Valoraci√≥n: 
          <span
            class="film-header__rate"
          />
        </p>
      </header>
      <section
        class="film-body"
      >
        <img
          alt="Poster de la pelicula undefined"
          class="film-body__image"
          src="https://image.tmdb.org/t/p/w342undefined"
        />
        <div
          class="film-body__data"
        >
          <h3
            class="film-body__title"
          />
          <p
            class="film-body__release"
          >
            Fecha de Estreno: 
            <span
              class="film-body__date"
            />
          </p>
          <p
            class="film-body__sipnosis"
          />
        </div>
      </section>
    </div>
  </main>
</DocumentFragment>
`;
